lib/Dialect/EmitC/Conversion/TosaToEmitC.cpp

```diff
@@ -108,16 +108,17 @@ class ConstOpConversion : public OpRewritePattern<tosa::ConstOp> {
   }
 };
 
-class Conv2DOpConversion : public OpConversionPattern<tosa::Conv2DOp> {
-  using OpConversionPattern<tosa::Conv2DOp>::OpConversionPattern;
+template <typename SrcOp>
+class Conv2DOpConversion : public OpConversionPattern<SrcOp> {
+  using OpConversionPattern<SrcOp>::OpConversionPattern;
 
 public:
-  Conv2DOpConversion(MLIRContext *ctx)
-      : OpConversionPattern<tosa::Conv2DOp>(ctx) {}
+  Conv2DOpConversion(MLIRContext *ctx, StringRef funcName)
+      : OpConversionPattern<SrcOp>(ctx), funcName(funcName) {}
 
 private:
   LogicalResult
-  matchAndRewrite(tosa::Conv2DOp conv2dOp, ArrayRef<Value> operands,
+  matchAndRewrite(SrcOp conv2dOp, ArrayRef<Value> operands,
                   ConversionPatternRewriter &rewriter) const override {
     // fail if quantization is requested
     if (conv2dOp.quantization_info().hasValue()) {
```

Comment:
```suggestion
    if (convOp.quantization_info().hasValue()) {
```

Reply:
This requires some further renaming at other places.

Reply:
Then also change the template parameter name to ConvOp.